server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCloseIndexAction.java

```diff
@@ -66,4 +70,24 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         return channel -> client.admin().indices().close(closeIndexRequest, new RestToXContentListener<>(channel));
     }
 
+    private String[] getIndicesFromRequest(final RestRequest request) throws IOException {
+        if (request.hasContent()) {
+            Map<String, Object> bodyMap = request.contentParser().map();
+            Object indicesObject = bodyMap.get("index");
+
+            if (indicesObject != null && request.hasParam("index")) {
+                throw new IllegalArgumentException("[index] specified both in path and in body. Specify it only once.");
+            } else if (indicesObject != null) {
+                return XContentMapValues.nodeStringArrayValue(indicesObject);
+            }
+        }
+
+        if (request.hasParam("index")) {
+            String indicesString = request.param("index");
+            return Strings.splitStringByCommaToArray(indicesString);
+        }
+
+        throw new IllegalArgumentException("_close should contain [index]. Specify it either in the path or in the body");
+    }
```

Comment:
Just to check, is it ok to put validation logic together with code logic? or would that be bad design?